src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -268,24 +268,37 @@ trait SubmissionComponent {
             val messages: Seq[AttributeString] = record.map {_.messageRecord}.collect { case Some(wm) => AttributeString(wm.message) }
 
             // attach the input resolutions to the workflow object
-            val workflowResolutions = groupedResolutions.get(wr.id).map {seq =>
-              seq.collect {
+            val workflowResolutions: Seq[SubmissionValidationValue] = groupedResolutions.get(wr.id).map { seq =>
+
+              //collect up the workflow resolution results by input
+              val resolutionsByInput = seq.collect {
                 case WorkflowAndInputResolutionRawSqlQuery.WorkflowInputResolutionListResult(workflow, Some(resolution), attribute) =>
-                  // unmarshalAttributes returns a highly nested structure; it is meant for bulk translation of
-                  // attributes, but we use it here for just one
-                  val attr = attribute.map{attrRec =>
-                    submissionAttributeQuery.unmarshalAttributes(Seq( ((attrRec.id,attrRec), None) ))
-                  }.map{_.values.head}.map{_.values.head}
-                SubmissionValidationValue(attr, resolution.errorText, resolution.inputName)
+                  (resolution, attribute)
+              }.groupBy { _._1.inputName }
+
+              //unmarshalAttributes will unmarshal multiple workflow attributes at once, but it expects all the attribute records to be real and not options.
```

Comment:
The solution is to marshal all the input resolutions for a single workflow in one go. `unmarshalAttributes` is capable of doing this, but expects all of the records it's marshalling to be real - and `WorkflowInputResolutionListResult` has them as `Option[SubmissionAttributeRecord]`s.

My approach here has therefore been to group the `WorkflowInputResolutionListResult` records by input resolution and unmarshal them one input at a time. This allows me to return an error if one input failed to resolve without affecting the others.